src/2d/shallow/surge/model_storm_module.f90

```diff
@@ -383,6 +383,92 @@ subroutine get_storm_data(t, storm, location, velocity, max_wind_radius,  &
     end subroutine get_storm_data
 
 
+    ! ==========================================================================
+    ! A set of common functions applied before calculating the circular component of 
+    ! wind speed using a particular wind model.
+ 
+    ! 1. Adjust max wind speed by removing translational speed (bounding at 0).
+
+    ! 2. Convert wind speed at 10 m to top of atmospheric boundary layer
+    ! ==========================================================================
+    subroutine adjust_max_wind(tv, mws, mod_mws)
```

Comment:
Maybe make this a `pure function` so that it can be inlined automatically?

Reply:
OK i added the pure modifier. Was that what you meant? Or is there something deeper. I wasn't able to figure out much about pure procedures online so I'm not really sure what I'm doing. I dont' think this can easily be made into a function? B/c it's modifying tv in addition to producing mod_mws

Reply:
Did it actually compile then?  I thought `pure` could only be added to a function.

Reply:
it did...but i'm wondering if it just ignored the pure modifier? Here's what I was using to try to figure out how pure works: http://www.lahey.com/docs/lfpro78help/F95ARPURE.htm. Suggests that it can be applied to subroutines, but seems like these ones don't actually meet the criteria